src/protojson/WorkerCzarComIssue.h

```diff
@@ -0,0 +1,131 @@
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef LSST_QSERV_PROTOJSON_WORKERCZARCOMISSUE_H
+#define LSST_QSERV_PROTOJSON_WORKERCZARCOMISSUE_H
+
+// System headers
+#include <chrono>
+#include <map>
+#include <memory>
+#include <set>
+#include <string>
+
+// Third party headers
+#include "nlohmann/json.hpp"
+
+// qserv headers
+#include "protojson/WorkerQueryStatusData.h"
+
+// This header declarations
+namespace lsst::qserv::protojson {
+
+/// This class is used to send/receive a message from the worker to a specific
+/// czar when there has been a communication issue with the worker sending UberJob
+/// file ready messages. If there have been timeouts, the worker will send this
+/// message to the czar immediately after the worker receives a
+/// WorkerQueryStatusData message from the czar (indicating that communication
+/// is now possible).
+/// If communication with the czar has failed for a long time, the worker
+/// will set "_thoughtCzarWasDead" and delete all incomplete work associated
+/// with that czar. Result files will remain until garbage cleanup or the czar
+/// calls for their removal.
+/// TODO:UJ  UberJob complete messages that failed to be sent to the czar
+/// TODO:UJ  will be added to this message. (uberjob file response)
+/// Upon successful completion, the worker will clear all values set by the
+/// the czar.
+/// Currently, this message is expected to only be needed rarely.
+class WorkerCzarComIssue {
+public:
+    using Ptr = std::shared_ptr<WorkerCzarComIssue>;
+
+    WorkerCzarComIssue() = delete;
+    ~WorkerCzarComIssue() = default;
+
+    std::string cName(const char* funcN) { return std::string("WorkerCzarComIssue") + funcN; }
+
+    static Ptr create(std::string const& replicationInstanceId_, std::string const& replicationAuthKey_) {
+        return Ptr(new WorkerCzarComIssue(replicationInstanceId_, replicationAuthKey_));
+    }
+
+    static Ptr createFromJson(nlohmann::json const& workerJson, std::string const& replicationInstanceId_,
+                              std::string const& replicationAuthKey_);
+
+    void setThoughtCzarWasDead(bool wasDead) {
+        std::lock_guard lg(_wciMtx);
+        _thoughtCzarWasDead = wasDead;
+    }
+
+    bool getThoughtCzarWasDead() const { return _thoughtCzarWasDead; }
+
+    /// Return true if there is a reason this WorkerCzarComIssue should be sent to this czar.
+    bool needToSend() const {
+        std::lock_guard lg(_wciMtx);
+        // TODO:UJ or list of failed transmits not empty.
+        return _thoughtCzarWasDead;
+    }
+
+    /// Set the contact information for the appropriate czar and worker.
+    void setContactInfo(WorkerContactInfo::Ptr const& wInfo_, CzarContactInfo::Ptr const& czInfo_) {
+        std::lock_guard lgWci(_wciMtx);
+        if (_wInfo == nullptr && wInfo_ != nullptr) _wInfo = wInfo_;
+        if (_czInfo == nullptr && czInfo_ != nullptr) _czInfo = czInfo_;
+    }
```

Comment:
Why have the online implementation of these methods in the header? This makes headers way too long and unreadable. Please note that in the modular programming paradigm (C++ and many other languages), _declarations_ is a way to define a contract of a class. Meanwhile, you would push all implementation details into the CC file. This makes your code readable and observable. Besides, it may as well speed up the compilation time of the project since headers may be included everywhere. More stuff you put into the header - longer the compilation could be.